packages/next-swc/crates/next-dev-tests/test-harness/harness.ts

```diff
@@ -210,3 +224,41 @@ export function markAsHydrated() {
     window.onNextHydrated()
   }
 }
+
+const fileChangedResolvers: Map<
+  string,
+  { resolve: (value: unknown) => void; reject: (error: Error) => void }
+> = new Map()
+
+globalThis.__turbopackFileChanged = (id: string, error?: Error) => {
+  const resolver = fileChangedResolvers.get(id)
+  if (resolver == null) {
+    throw new Error(`No resolver found for id ${id}`)
+  } else if (error != null) {
+    resolver.reject(error)
+  } else {
+    resolver.resolve(null)
+  }
+}
+
+function unsafeUniqueId(): string {
+  const LENGTH = 10
+  const BASE = 16
+  return Math.floor(Math.random() * Math.pow(BASE, LENGTH))
+    .toString(BASE)
+    .slice(0, LENGTH)
+}
+
+export async function changeFile(
+  path: string,
+  find: string,
+  replaceWith: string
+) {
+  return new Promise((resolve, reject) => {
+    const id = unsafeUniqueId()
```

Comment:
Nit: We could loop to make this safe:
```suggestion
    let id;
    while ((id = unsafeUniqueId())) {
      if (!fileChangedResolvers.has(id)) break;
    }
```